### PR TITLE
Feature/service integration

### PR DESCRIPTION
**Pull request by freekvandeven**
This now adds template and availability loading on the availability overview.
In the local data interface i added some mocked data for the current month for easy testing.
On the template overview page it now loads all the templates and you can do all crud operations on them.
The calendar itself doesn't show loading state yet but this will be handled at a later point

### CODE REVIEW

File: packages/flutter_availability/test/util/availability_deviation_test.dart [hunk 1]
```diff
@@ -0,0 +1,156 @@
+import "package:flutter_availability/src/util/availability_deviation.dart";
+import "package:flutter_availability_data_interface/flutter_availability_data_interface.dart";
+import "package:flutter_test/flutter_test.dart";
+
+void main() {
+  group("isTemplateDeviated", () {
+    test("returns false when template is null", () {
```
Comment (PureTryOut) [inline packages/flutter_availability/test/util/availability_deviation_test.dart]: I'm not sure this is the result you want. How can it be marked "deviated" if there is no template to compare too? I wouldn't accept a null value in the first place, it makes no sense.

File: packages/flutter_availability/test/util/availability_deviation_test.dart [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+          startTime: DateTime(2023, 7, 9, 9, 0),
+          endTime: DateTime(2023, 7, 9, 17, 0),
+          breaks: [],
+        ),
+      );
+      var result = isTemplateDeviated(availability, template);
+      expect(result, isTrue);
+    });
+
+    test("returns false when availability matches the week template", () {
```
Comment (PureTryOut) [inline packages/flutter_availability/test/util/availability_deviation_test.dart]: It would be easier to read if you put day and week testing in separate groups. For a moment I was wondering why you were testing the same thing twice.

File: packages/flutter_availability/test/util/availability_deviation_test.dart [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            breaks: [],
+          ),
+        ),
+      );
+      var result = isTemplateDeviated(availability, template);
+      expect(result, isTrue);
+    });
+
+    test(
+        "handles edge cases where template data is missing "
```
Comment (PureTryOut) [inline packages/flutter_availability/test/util/availability_deviation_test.dart]: No need to mention "edge case" in your tests. With the current title you're not describing the result you want.

Something like `returns true when the template is missing the day of the availability` would be better.

File: packages/flutter_availability/test/util/availability_deviation_test.dart [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        "for a day in week template", () {
+      var availability = AvailabilityModel(
+        userId: "user1",
+        startDate: DateTime(2023, 7, 10, 10, 0),
```
Comment (PureTryOut) [inline packages/flutter_availability/test/util/availability_deviation_test.dart]: I don't know what day in the week this is, maybe add that information as a comment. I suppose this is Monday since it's a day later than the Sunday in the template?

File: packages/flutter_availability/test/util/availability_deviation_test.dart [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+        breaks: [],
+      );
+      var template = AvailabilityTemplateModel(
+        userId: "user1",
+        name: "Week Template",
+        color: 0,
+        templateType: AvailabilityTemplateType.week,
+        templateData: WeekTemplateData(
+          data: {
+            WeekDay.sunday: DayTemplateData(
```
Comment (PureTryOut) [inline packages/flutter_availability/test/util/availability_deviation_test.dart]: Why are you providing a day at all if you're not testing against it? Can't you just pass an empty dictionary?

**Review by PureTryOut — CHANGES_REQUESTED** [review]

**Review by PureTryOut — APPROVED** [review]
